core/src/test/groovy/com/optum/sourcehawk/core/repository/LocalRepositoryFileWriterSpec.groovy

```diff
@@ -0,0 +1,107 @@
+package com.optum.sourcehawk.core.repository
+
+import spock.lang.Specification
+
+import java.nio.charset.StandardCharsets
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.util.stream.Collectors
+
+class LocalRepositoryFileWriterSpec extends Specification {
+
+    def "create"() {
+        expect:
+        LocalRepositoryFileWriter.writer()
+    }
+
+    def "write text to file and read it"() {
+        given:
+        String fileName = UUID.randomUUID().toString()
+        File fileResource = File.createTempFile(fileName, ".txt")
+        Path repositoryRoot = Paths.get(fileResource.getParentFile().getAbsolutePath())
+        RepositoryFileWriter repositoryFileWriter = LocalRepositoryFileWriter.writer()
+        RepositoryFileReader repositoryFileReader = LocalRepositoryFileReader.create(repositoryRoot)
+
+        when:
+        Optional<InputStream> fileInputStream = repositoryFileReader.read(fileResource.getAbsoluteFile().toString())
+        String fileText = new BufferedReader(new InputStreamReader(fileInputStream.get(), StandardCharsets.UTF_8)).lines().collect(Collectors.joining("\n"))
+
+        then:
+        fileInputStream.isPresent()
+        fileText.isEmpty()
+
+        when:
+        repositoryFileWriter.write(fileResource.getAbsolutePath(), "hello".bytes)
+
+        then:
+        notThrown(Exception)
+
+        when:
+        fileInputStream = repositoryFileReader.read(fileResource.getAbsolutePath())
+        String helloFileText = new BufferedReader(new InputStreamReader(fileInputStream.get(), StandardCharsets.UTF_8)).lines().collect(Collectors.joining("\n"))
```

Comment:
You can use groovy's `InputStream.text` here to simplify

Reply:
Thanks